src/widgets/signature/SignatureWidget.js

```diff
@@ -43,70 +41,55 @@ const ClearButton = styled(IconButton)`
 `;
 
 type Props = {
-  disabled :boolean,
-  value :string,
-  onChange :(data :string | void) => void
+  disabled :boolean;
+  onChange :(data :string | void) => void;
+  value :?string;
 };
 
-class SignatureWidget extends Component<Props> {
-
-  signatureCanvasRef :{ current :?SignatureCanvas } = React.createRef();
-
-  componentDidUpdate(prevProps :Props) {
-    const { disabled, value } = this.props;
-    const { current } = this.signatureCanvasRef;
-    if (disabled && current) {
-      current.off();
-    }
-    else if (current) {
-      current.on();
-    }
+const SignatureWidget = (props :Props) => {
 
-    if (isNonEmptyString(value)) {
-      const newCanvasValue :string = tryConvertingToDataURL(value);
-      const prevCanvasValue :string = tryConvertingToDataURL(prevProps.value);
-      if (newCanvasValue !== prevCanvasValue) {
-        this.setCanvas(newCanvasValue);
-      }
-    }
-  }
+  const { disabled, onChange, value } = props;
+  const signatureCanvasRef :{ current :?SignatureCanvas } = useRef();
 
-  handleClear = () => {
-    const { onChange } = this.props;
-    const { current } = this.signatureCanvasRef;
+  useEffect(() => {
+    const { current } = signatureCanvasRef;
     if (current) {
-      current.clear();
-      onChange();
+      if (disabled) {
+        current.off();
+      }
+      else {
+        current.on();
+      }
+      if (value) {
+        const valueAsDataURL = tryConvertingToDataURL(value);
+        current.fromDataURL(valueAsDataURL);
+      }
     }
-  }
+  }, [disabled, signatureCanvasRef, value]);
 
-  setCanvas = (data :string) => {
-    const { current } = this.signatureCanvasRef;
+  const handleOnChange = useCallback(() => {
+    const { current } = signatureCanvasRef;
     if (current) {
-      current.fromDataURL(data);
+      // https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL
+      const dataURL :string = current.toDataURL(); // default is "image/png"
+      onChange(dataURL.slice(DATA_URL_IMG_PNG_PREFIX.length));
     }
-  }
+  }, [onChange, signatureCanvasRef]);
 
-  handleChange = () => {
-    const { onChange } = this.props;
-    const { current } = this.signatureCanvasRef;
+  const handleOnClear = useCallback(() => {
+    const { current } = signatureCanvasRef;
     if (current) {
-      // https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL
-      const dataUrl :string = current.toDataURL(); // default is "image/png"
-      onChange(dataUrl.slice(DATA_URL_IMG_PNG_PREFIX.length));
+      current.clear();
+      onChange();
     }
-  }
-
-  render() {
-
-    const { disabled } = this.props;
-    return (
-      <CanvasWrapper>
-        <SignatureCanvas onEnd={this.handleChange} ref={this.signatureCanvasRef} />
-        <ClearButton disabled={disabled} icon={ClearIcon} mode="subtle" onClick={this.handleClear} />
-      </CanvasWrapper>
-    );
-  }
-}
+  }, [onChange, signatureCanvasRef]);
```

Comment:
Does passing `signatureCanvasRef.current` work? Could maybe avoid the sanity check inside.